projects/packages/search/src/instant-search/class-instant-search.php

```diff
@@ -309,97 +311,161 @@ public function get_search_aggregations_results() {
 		}
 
-		$next_id         = 1;
 		$widget_opt_name = Helper::get_widget_option_name();
 		$widget_options  = get_option( $widget_opt_name, array() );
-		foreach ( $widget_options as $id => $w ) {
-			$id = intval( $id );
-			if ( $id >= $next_id ) {
-				$next_id = $id + 1;
-			}
-		}
 
-		// Copy sidebar settings to overlay.
-		if ( ( false !== $sidebar_id ) && isset( $widget_options[ $sidebar_id ] ) ) {
-			$widget_options[ $next_id ] = $widget_options[ $sidebar_id ];
-			update_option( $widget_opt_name, $widget_options );
+		$next_id = $this->get_next_jp_search_widget_id( $widget_options );
 
-			if ( ! isset( $sidebars['jetpack-instant-search-sidebar'] ) ) {
-				$sidebars['jetpack-instant-search-sidebar'] = array();
-			}
-			array_unshift( $sidebars['jetpack-instant-search-sidebar'], Helper::build_widget_id( $next_id ) );
-			update_option( 'sidebars_widgets', $sidebars );
+		list(,$sidebar_jp_searchbox_wiget_id ) = $this->get_search_widget_indices( $sidebars, self::AUTO_CONFIG_SIDEBAR );
+		if ( false !== $sidebar_jp_searchbox_wiget_id && isset( $widget_options[ $sidebar_jp_searchbox_wiget_id ] ) ) {
+			// If there is a JP search widget in the theme sidebar, copy it over to the search overlay sidebar.
+			$widget_options[ $next_id ] = $widget_options[ $sidebar_jp_searchbox_wiget_id ];
+		} else {
+			// If JP Search widget doesn't exist in the theme sidebar, we have nothing to copy from, so we create a new one within the overlay sidebar.
+			$widget_options[ $next_id ] = $this->get_preconfig_widget_options();
+		}
+		array_unshift( $sidebars['jetpack-instant-search-sidebar'], Helper::build_widget_id( $next_id ) );
+		update_option( $widget_opt_name, $widget_options );
+		update_option( 'sidebars_widgets', $sidebars );
+		return true;
+	}
 
+	/**
+	 * Add JP Search widget on top of theme sidebar.
+	 * Or Replace core search widget in theme sidebar if exists.
+	 */
+	public function auto_config_theme_sidebar_search_widget() {
+		$sidebars = get_option( 'sidebars_widgets', array() );
+		if ( ! isset( $sidebars[ self::AUTO_CONFIG_SIDEBAR ] ) ) {
 			return;
 		}
 
-		// Configure overlay and sidebar (if it exists).
-		$preconfig_opts = $this->get_preconfig_widget_options();
-		if ( ! isset( $sidebars['jetpack-instant-search-sidebar'] ) ) {
-			$sidebars['jetpack-instant-search-sidebar'] = array();
-		}
-		if ( $has_sidebar ) {
-			$widget_options[ $next_id ] = $preconfig_opts;
-			if ( false !== $sidebar_searchbox_idx ) {
-				// Replace Core search box.
-				$sidebars['sidebar-1'][ $sidebar_searchbox_idx ] = Helper::build_widget_id( $next_id );
-			} else {
-				// Add to top.
-				array_unshift( $sidebars['sidebar-1'], Helper::build_widget_id( $next_id ) );
+		// If there's JP search widget in theme sidebar, skip.
+		foreach ( (array) $sidebars[ self::AUTO_CONFIG_SIDEBAR ] as $widget_id ) {
+			if ( 0 === strpos( $widget_id, Helper::FILTER_WIDGET_BASE ) ) {
+				// Already configured.
+				return;
 			}
-			$next_id++;
 		}
```

Comment:
Like with `auto_config_overlay_sidebar_widgets`, can we replace this by checking if the second value of `$this->get_search_widget_indices( $sidebars )` (L384) is false?

Reply:
Addressed in 352d1ad86f